pkg/prometheus/rules.go

```diff
@@ -221,14 +224,37 @@ func (c *Operator) selectRules(p *monitoringv1.Prometheus, namespaces []string)
 	return rules, nil
 }
 
-func generateContent(promRule monitoringv1.PrometheusRuleSpec) (string, error) {
+func generateContent(promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
 	content, err := yaml.Marshal(promRule)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to marshal content")
 	}
+	errs := validateRule(promRule)
+	if len(errs) != 0 {
+		const m = "Invalid rule"
+		level.Debug(logger).Log("msg", m, "content", content)
+		for _, err := range errs {
+			level.Info(logger).Log("msg", m, "err", err)
+		}
+		return "", errors.New(m)
+	}
```

Comment:
Would it make sense to wrap it into a function and reuse in `pkg/thanos/rules.go` as well as in admission webhook code?